internal/cloudfront/origin.go

```diff
@@ -56,23 +70,34 @@ type Behavior struct {
 
 // OriginBuilder allows the construction of a Origin
 type OriginBuilder struct {
-	host          string
-	viewerFnARN   string
-	requestPolicy string
-	cachePolicy   string
-	respTimeout   int64
-	paths         strhelper.Set
+	host             string
+	viewerFnARN      string
+	requestPolicy    string
+	distributionName string
+	cachePolicy      string
+	respTimeout      int64
+	originType       string
```

Comment:
Yeah, I initially thought of mapping the concepts to the terms we were using in our discussion (usually we were saying "a bucket origin" and similar). But thinking more about this I think it would be better to use the AWS concept. Thanks for pointing this out o/